app/src/main/kotlin/com/wire/android/ui/home/conversations/edit/EditMessageMenuItems.kt

```diff
@@ -125,70 +124,117 @@ fun EditMessageMenuItems(
         }
     }
 
-    return buildList {
-        if (isAvailable) {
-            add { ReactionOption(onReactionItemClick) }
-            add { MessageDetailsMenuOption(onDetailsItemClick) }
-            if (isCopyable) {
-                add {
-                    MenuBottomSheetItem(
-                        icon = {
-                            MenuItemIcon(
-                                id = R.drawable.ic_copy,
-                                contentDescription = stringResource(R.string.content_description_copy_the_message),
-                            )
-                        },
-                        title = stringResource(R.string.label_copy),
-                        onItemClick = onCopyItemClick
-                    )
+    if (message.expirationStatus is ExpirationStatus.Expirable) {
+        return EphemeralMessageEditMenuItems(
+            message = message,
+            onDetailsClick = onDetailsItemClick,
+            onDownloadAsset = onDownloadAssetClick,
+            onOpenAsset = onOpenAssetClick,
+            onDeleteMessage = onDeleteItemClick
+        )
+    } else {
+        return buildList {
+            if (isAvailable) {
+                add { ReactionOption(onReactionItemClick) }
+                add { MessageDetailsMenuOption(onDetailsItemClick) }
+                if (isCopyable) {
+                    add {
+                        MenuBottomSheetItem(
+                            icon = {
+                                MenuItemIcon(
+                                    id = R.drawable.ic_copy,
+                                    contentDescription = stringResource(R.string.content_description_copy_the_message),
+                                )
+                            },
+                            title = stringResource(R.string.label_copy),
+                            onItemClick = onCopyItemClick
```

Comment:
Since this class started growing quite a bit, can we move this `MenuBottomSheetItem` to its own file, aka `CopyTextMenuOption`?

Reply:
I moved them to a sepearte private function inside this class fiel I think it is better to have it only there since it is used only there and you do not need to jump between the files, let me know if that is fine with you by resolving